src/detours.cpp

```diff
@@ -2545,4 +2545,20 @@ BOOL WINAPI DetourVirtualProtectSameExecute(_In_  PVOID pAddress,
                                              pAddress, nSize, dwNewProtect, pdwOldProtect);
 }
 
+BOOL WINAPI DetourAreSameGuid(_In_ REFGUID left, _In_ REFGUID right)
+{
+    return
```

Comment:
I realize  this is just a nice refactor of the existing code, but is there a reason not use IsEqualGUID that I didn't notice?

Reply:
Using `IsEqualGUID` makes detours depend on ole32.lib.

Reply:
Uhh.. maybe they changed it in different versions of the headers...

In my windows SDK (version 10.0.19041.0) `IsEqualGUID` is declared as a macro or inline function depending on how you are compiling. 

```cpp

#if !defined (__midl)
#if !defined (_SYS_GUID_OPERATORS_)
#define _SYS_GUID_OPERATORS_
#include <string.h>

// Faster (but makes code fatter) inline version...use sparingly
#ifdef __cplusplus
__inline int InlineIsEqualGUID(REFGUID rguid1, REFGUID rguid2)
{
   return (
      ((unsigned long *) &rguid1)[0] == ((unsigned long *) &rguid2)[0] &&
      ((unsigned long *) &rguid1)[1] == ((unsigned long *) &rguid2)[1] &&
      ((unsigned long *) &rguid1)[2] == ((unsigned long *) &rguid2)[2] &&
      ((unsigned long *) &rguid1)[3] == ((unsigned long *) &rguid2)[3]);
}

__inline int IsEqualGUID(REFGUID rguid1, REFGUID rguid2)
{
    return !memcmp(&rguid1, &rguid2, sizeof(GUID));
}

#else   // ! __cplusplus

#define InlineIsEqualGUID(rguid1, rguid2)  \
        (((unsigned long *) rguid1)[0] == ((unsigned long *) rguid2)[0] &&   \
        ((unsigned long *) rguid1)[1] == ((unsigned long *) rguid2)[1] &&    \
        ((unsigned long *) rguid1)[2] == ((unsigned long *) rguid2)[2] &&    \
        ((unsigned long *) rguid1)[3] == ((unsigned long *) rguid2)[3])

#define IsEqualGUID(rguid1, rguid2) (!memcmp(rguid1, rguid2, sizeof(GUID)))

#endif  // __cplusplus
```

Reply:
Ah, I was going off the [documentation](https://docs.microsoft.com/en-us/windows/win32/api/guiddef/nf-guiddef-isequalguid)... We just need to know since when that was made inline. For example, I'm not sure if whatever SDK supported by VS 2005 also does that :/

Reply:
Sure, it looks like it still exports IsEqualGUID... 
```
$ dumpbin /exports C:\Windows\System32\ole32.dll | rg IsEqualGUID
       1139  140 00010CA0 IsEqualGUID
```

I think your change is correct, maybe we should just add a comment to
DetourAreSameGuid, explaining that it exists for backwards compat.

Reply:
I added a comment